drivers/hmis/app/models/hmis/form/form_processor.rb

```diff
@@ -69,16 +69,46 @@ def run!(owner:, user:)
       end
     end
 
-    # Iterate through each used processor to apply metadata and information dates
-    relevant_container_names = hud_values.keys.map { |k| parse_key(k)&.first }.compact.uniq
-    relevant_container_names.each do |container|
-      container_processor(container)&.assign_metadata
-      container_processor(container)&.information_date(custom_assessment.assessment_date) if custom_assessment.present?
+    # Iterate through each container (record type) to apply metadata and information dates
+    hud_values_by_container.keys.each do |container|
+      processor = container_processor(container)
+
+      # If this is an assessment and all fields pertaining to this record type were hidden,
+      # the related record should be destroyed. (For example, a Custom Assessment that conditionally creates a CE Event).
+      if custom_assessment.present? && containers_with_all_fields_hidden.include?(container) && !['Assessment', 'Enrollment'].include?(container)
```

Comment:
Is there a better way to write this? 
```
!['Assessment', 'Enrollment'].include?(container)
```

Can you ask the container if it's custom or something?  Just trying to think about the future and what other classes may have required dependent items, seems somewhat easier to follow if the container can answer that question.

Reply:
Yesss. I put this in at the last minute because I got afraid of unintentional deletions. I think it would be better if we mark on the processor whether its ALLOWED to be a conditional record. Then we can just turn it on for CE Event, Ce Assmt, and CLS.

Reply:
could this be implemented as:

```
container.dependent_destroyable?
``` 

or similar?

Reply:
If this is just an extra safety check, I suggest it should be an exception rather than part of the condition. something like:
```ruby
if custom_assessment.present? && containers_with_all_fields_hidden.include?(container) 
 raise "unexpected container #{container}" if ['Assessment', 'Enrollment'].include?(container)
 processor.destroy_record
```